---
title: Observable.Catch Method  (System.Reactive.Linq)
TOCTitle: Catch Method
ms:assetid: Overload:System.Reactive.Linq.Observable.Catch
ms:mtpsurl: https://msdn.microsoft.com/en-us/library/system.reactive.linq.observable.catch(v=VS.103)
ms:contentKeyID: 36068357
ms.date: 06/28/2011
mtps_version: v=VS.103
f1_keywords:
- System.Reactive.Linq.Observable.Catch
- System.Reactive.Linq.Observable.Catch``1
- System.Reactive.Linq.Observable.Catch``2
dev_langs:
- CSharp
- JScript
- VB
- FSharp
---

# Observable.Catch Method

Include Protected Members  
Include Inherited Members  

Continues an observable sequence that is terminated by an exception.

This member is overloaded. For complete information about this member, including syntax, usage, and examples, click a name in the overload list.

## Overload List

<table>
<thead>
<tr class="header">
<th> </th>
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><img src="images\Hh303103.pubmethod(en-us,VS.103).gif" title="Public method" alt="Public method" /><img src="images\Hh244319.static(en-us,VS.103).gif" title="Static member" alt="Static member" /></td>
<td><a href="https://msdn.microsoft.com/en-us/library/m:system.reactive.linq.observable.catch%60%601(system.collections.generic.ienumerable%7bsystem.iobservable%7b%60%600%7d%7d)(v=VS.103)">Catch&lt;TSource&gt;(IEnumerable&lt;IObservable&lt;TSource&gt;&gt;)</a></td>
<td>Continues an observable sequence that is terminated by an exception with the next observable sequence.</td>
</tr>
<tr class="even">
<td><img src="images\Hh303103.pubmethod(en-us,VS.103).gif" title="Public method" alt="Public method" /><img src="images\Hh244319.static(en-us,VS.103).gif" title="Static member" alt="Static member" /></td>
<td><a href="https://msdn.microsoft.com/en-us/library/m:system.reactive.linq.observable.catch%60%601(system.iobservable%7b%60%600%7d%5b%5d)(v=VS.103)">Catch&lt;TSource&gt;(array&lt;IObservable&lt;TSource&gt;[])</a></td>
<td>Continues an observable sequence that is terminated by an exception with the next observable sequence.</td>
</tr>
<tr class="odd">
<td><img src="images\Hh303103.pubmethod(en-us,VS.103).gif" title="Public method" alt="Public method" /><img src="images\Hh244319.static(en-us,VS.103).gif" title="Static member" alt="Static member" /></td>
<td><a href="https://msdn.microsoft.com/en-us/library/m:system.reactive.linq.observable.catch%60%602(system.iobservable%7b%60%600%7d%2csystem.func%7b%60%601%2csystem.iobservable%7b%60%600%7d%7d)(v=VS.103)">Catch&lt;TSource, TException&gt;(IObservable&lt;TSource&gt;, Func&lt;TException, IObservable&lt;TSource&gt;&gt;)</a></td>
<td>Continues an observable sequence that is terminated by an exception of the specified type with the observable sequence produced by the handler.</td>
</tr>
<tr class="even">
<td><img src="images\Hh303103.pubmethod(en-us,VS.103).gif" title="Public method" alt="Public method" /><img src="images\Hh244319.static(en-us,VS.103).gif" title="Static member" alt="Static member" /></td>
<td><a href="https://msdn.microsoft.com/en-us/library/m:system.reactive.linq.observable.catch%60%601(system.iobservable%7b%60%600%7d%2csystem.iobservable%7b%60%600%7d)(v=VS.103)">Catch&lt;TSource&gt;(IObservable&lt;TSource&gt;, IObservable&lt;TSource&gt;)</a></td>
<td>Continues an observable sequence that is terminated by an exception with the next observable sequence.</td>
</tr>
</tbody>
</table>

Top

## See Also

#### Reference

[Observable Class](hh244252\(v=vs.103\).md)

[System.Reactive.Linq Namespace](hh211929\(v=vs.103\).md)

